ambry-replication/src/main/java/com.github.ambry.replication/ReplicationEngine.java

```diff
@@ -141,7 +147,25 @@ public void updateTotalBytesReadByRemoteReplica(PartitionId partitionId, String
       long totalBytesRead) {
     RemoteReplicaInfo remoteReplicaInfo = getRemoteReplicaInfo(partitionId, hostName, replicaPath);
     if (remoteReplicaInfo != null) {
+      ReplicaId localReplica = remoteReplicaInfo.getLocalReplicaId();
       remoteReplicaInfo.setTotalBytesReadFromLocalStore(totalBytesRead);
+      // update replication lag in ReplicaSyncUpManager
+      if (replicaSyncUpManager != null) {
+        Store localStore = storeManager.getStore(partitionId);
+        if (localStore.getCurrentState() == ReplicaState.INACTIVE) {
+          // TODO incorporate changes for Standby-To-Inactive transition.
+        } else if (localStore.isDecommissionInProgress() && localStore.getCurrentState() == ReplicaState.OFFLINE) {
+          // if local store is in OFFLINE state, we need more info to determine if replica is really in Inactive-To-Offline
+          // transition. So we check if decommission file is present. If present, we update SyncUpManager by peer's lag
+          // from end offset in local store.
+          boolean updated =
+              replicaSyncUpManager.updateLagBetweenReplicas(localReplica, remoteReplicaInfo.getReplicaId(),
+                  localStore.getSizeInBytes() - totalBytesRead);
```

Comment:
I am confused by the calculation here. localStore.getSizeInBytes() might actually have more bytes than the totalBytesRead since we ignore the delete blob in replication. Is this an accurate difference or just a estimate?

Reply:
It's actually accurate. `totalBytesRead` is calculated in metadata exchange phase and is computed based on remote replica token (offset in this token). The bytes read by remote replica contain all the records in local store (PUT/DELETE/TTL Update), although remote replica is able to drop the delete record if original put record is not found in its store.